src/docfx/build/context/Output.cs

```diff
@@ -95,6 +95,15 @@ public void Delete(string destRelativePath)
             {
                 File.Delete(destinationPath);
             }
+
+            if (legacy)
+            {
+                var mtaJsonPath = Path.ChangeExtension(destinationPath.Replace(".raw.page.json", ".json"), "mta.json");
```

Comment:
why not directly repleace the extension?

Reply:
There could be other formats like `.png` for resource and `.json` for TOC.

Reply:
it's the Delete is deleting multiple files instead of one file, why not passing in the publish item, using it to decide which files need to be delete from file system, in case we have more in the future